src/params/progression.jl

```diff
@@ -34,17 +42,38 @@ function sample_severity(rng::AbstractRNG, age::Real)
   elseif age < 80;  dist = severity_dists[5]
   else;             dist = severity_dists[6]
   end
+  idx = gender + 1 |> Int32
+  prob = hospitalization_probs[idx][age < max_age_hosp ? age + 1 : max_age_hosp]
+  if age < 0;       error("age should be non-negative")
+  elseif age < 40;  dist = Categorical([0,  1.0 - prob,  prob *(1-critical_probs[1]) ,  prob *critical_probs[1]])
```

Comment:
maybe if we have `const age_groups = [40, 50, 60, 70, 80]`, then we can apply `for` loop or something :)

Reply:
This looks like a use case for `MatrixSampler` it does exactly this some groups and a discriete distribution of the same length for each group. It needs a matrix where each column is one distribution. The sample function needs `source` which essentially is the number of column to use

Reply:
For the grouping you can use the `agegroup` function and include the gender

Reply:
ok